app/components/submission/ManuscriptMetadata/SubjectAreaDropdown.test.js

```diff
@@ -0,0 +1,120 @@
+import React from 'react'
+import { shallow, mount } from 'enzyme'
+import theme from '@pubsweet/elife-theme'
+
+import SubjectAreaDropdown from './SubjectAreaDropdown'
+
+const onChange = jest.fn()
+
+const MyDropdown = (
+  <SubjectAreaDropdown
+    label="My label"
+    name="My name"
+    onBlur={jest.fn()}
+    onChange={onChange}
+    savedValues={[]}
+    theme={theme}
+  />
+)
+
+describe('SubjectAreaDropdown component', () => {
+  it('renders label & select', () => {
+    const wrapper = shallow(MyDropdown).dive() // dive is necessary because SubjectAreaDropdown is wrapped by withTheme
+    expect(wrapper.contains('My label')).toBe(true)
+    expect(wrapper.find('[inputId="subject-area-select"]').exists()).toEqual(
+      true,
+    )
+    expect(wrapper.contains('No more than two subject areas')).toBe(false)
+  })
+
+  describe('Integration with react-select', () => {
```

Comment:
Do we care about `react-select` at test level?

Reply:
why not?

Reply:
Again, implementation detail. I don't care that the SubjectAreaDropdown is using `react-select`, I just care if it works or not.

Reply:
True, it is an implementation detail. But every test that's written here assumes an HTML structure that is specific to `react-select`. When testing the integration between components, I assume that specifying which components you're using is fine?

Reply:
Perhaps this wasn't your point? Are you actually commenting on the choice to write integration tests & not really do unit tests?

I've got to say, the lack of unit tests feels a little wrong to me too. But I asked about this when I first started working on testing for this component. It seemed to be a commonly held opinion that integration tests would add more value than unit tests & that unit tests may not even be worth writing. I'd be happy to raise a discussion issue, for defining a testing strategy for the frontend?

Regardless of whether we choose to add unit tests though, integration tests are very valuable. At the end of the day, you want to be sure that the unit-tested functionality of `react-select` combined with the unit tested functionality of SubjectAreaDropdown, give you the behaviour that you want to see.

Does this answer your point?

Reply:
No, I was just saying I don't agree with how you named your describe section. I would name it differently.